extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnWriterImpl.java

```diff
@@ -122,7 +124,12 @@ public void addDictionaryPage(final Object dictionaryValues, final int valuesCou
     public void writeDictionaryPage(final ByteBuffer dictionaryBuffer, final int valuesCount) throws IOException {
         long currentChunkDictionaryPageOffset = writeChannel.position();
         int uncompressedSize = dictionaryBuffer.remaining();
-        BytesInput compressedBytes = compressor.compress(BytesInput.from(dictionaryBuffer));
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try (WritableByteChannel channel = Channels.newChannel(compressor.compress(baos))) {
+            channel.write(dictionaryBuffer);
+        }
+        BytesInput compressedBytes = BytesInput.from(baos);
```

Comment:
Any reason we don't turn this into a re-usable method? We do it 3 times in this file alone.

Reply:
I thought there was a reason, I'll double check and confirm. I know at least one instance _doesn't_ match, and might have just made the bad decision to leave the others alone.

Reply:
There are two, which is below my personal threshold for "refactor a method for three lines of code" - the third is not on a bytebuffer being written, but a bytesinput - the bb makes sense as roughly `out.write(bb)` inside the try-with-resources, whereas the output stream would be `bytesinput.writeAllTo(out)`. 

Technically we could wrap both ByteBuffers up in a BytesInput, but unwrapping the BytesInput would result in dealing with between 1 and 3 ByteBuffers. My vote is that it doesn't generalize well enough for this to be worth it.